util/dvsim/LocalLauncher.py

```diff
@@ -74,6 +84,9 @@ def poll(self):
         '''
 
         assert self.process is not None
+        if self.timeout_secs:
+            if self.start_time + self.timeout_secs < datetime.datetime.now():
+                self.kill()
```

Comment:
IMO job killed due to timeout should reflect an appropriate error message, such as `f"Job killed due to {self.job_timeout_mins} minute timeout being exceeded."`. 

Consider moving the lines 107-113 in `kill()` method below to `_kill()`. So `_kill()` will only invoke the `self.process.terminate()`, without invoking the `_post_finish()` method. Then you can invoke the `_kill()` here, followed by `_post_finish()` method and set a more accurate error message.

Reply:
Great suggestion, and since the message will be more accurate there is no need to log that a timeout is in effect (line 55).

Reply:
By the way, we should return 'K' in this case, and update the function doc string? The Scheduler accepts a return of 'K'.

Reply:
Yes, that would be the right way to do it.

Reply:
Done